### PR TITLE
Jc/add name to users

### PR DESCRIPTION
**Pull request by jcobert**
I added first and last name fields to the users schema. I also adjusted the email input validation to require a domain after the @ sign.

### CODE REVIEW

**Comment by jmagardino** [comment]
once we merge the other PR, we can pull master into this and resolve some conflicts.

File: lib/kg/accounts/user.ex [hunk 1]
```diff
@@ -26,15 +28,19 @@ defmodule Kg.Accounts.User do
...
+>>>>>>> 0a9e7ac537cee73466e35cd7bda4950ee7e2a793
     |> validate_email()
     |> validate_password()
   end
 
   defp validate_email(changeset) do
     changeset
     |> validate_required([:email])
-    |> validate_format(:email, ~r/^[^\s]+@[^\s]+$/, message: "must have the @ sign and no spaces")
+    |> validate_format(:email, ~r/^[^\s]+@[^\s]+\.+[^\s]+$/, message: "Please enter a valid email")
```
Comment (jmagardino) [inline lib/kg/accounts/user.ex:39]: Are we overwriting an old change by accident or is this what you want?

**Review by jmagardino — CHANGES_REQUESTED** [review]
You said 
`I also adjusted the email input validation to require a domain after the @ sign.`

Can we add a test that ensures that validation happens?

File: lib/kg/accounts/user.ex (hunk 1 above)
Comment (jcobert) [reply lib/kg/accounts/user.ex:39]: That was intended. Bad idea? I noticed that a user could sign up using myemail@nodomain so I thought that would help.

Comment (jmagardino) [reply lib/kg/accounts/user.ex:39]: nope I misread the code, that sounds good 👍 

This error message gets passed down into the view, so some of the tests now fail. I'll talk to you in iMessage to explain how to run the tests.

File: test/kg/accounts_test.exs [hunk 2]
```diff
@@ -92,6 +92,16 @@ defmodule Kg.AccountsTest do
       assert is_nil(user.password)
     end
 
+    test "registers users with a first and last name" do
+      email = unique_user_email()
+      first_name = "Tim"
+      last_name = "Doyle"
+      {:ok, user} = Accounts.register_user(%{email: email, password: valid_user_password()})
```
Comment (jmagardino) [inline test/kg/accounts_test.exs:99]: You defined `first_name` and `last_name` variables which is a good first start. But you need to add these into the parameters to register_user. You'll see that this test fails when running `mix test`

**Comment by jmagardino** [comment]
Closing this PR as you fixed everything in the other PR

**Pull request closed without merging**